packages/node_modules/@ciscospark/internal-plugin-board/src/board.js

```diff
@@ -147,10 +149,31 @@ const Board = SparkPlugin.extend({
       aclLinkOperation: `DELETE`
     };
 
+    let promise = Promise.resolve();
+    if (options.preventDeleteActiveChannel) {
+      promise = this.lockChannelForDeletion(channel);
```

Comment:
Why does this need to be done on the client? Why doesn't the service send a 409 if the channel is in use?

Reply:
The current delete (aka remove channel from a space) is done by the client sending `remove ACL Links` to KMS directly. The board service does not have control over KMS so this `/lock?intent=delete` from the board service is a helper for clients.